http-server/src/access_control/cors.rs

```diff
@@ -49,22 +49,22 @@ pub enum OriginProtocol {
 pub struct Origin {
 	protocol: OriginProtocol,
 	host: Host,
-	as_string: String,
+	host_with_proto: String,
```

Comment:
almost like we should refactor this to a separate type ^^ :)

I think this entire `access_control module` needs to checked a little bit more carefully, I don't remember but this matcher stuff could maybe be replaced by the `url crate` where they have functions for parsing url related stuff.